### PR TITLE
sanity, fixes and qols for gargoyles

### PR DESCRIPTION
**Pull request by Seris02**
# About The Pull Request
so I started playing a gargoyle because I thought it would be fun and noticed these things, so here's the pr to fix them.

as title says
- removes white underlay from the gargoyle (ie, the normal gargoyle statue sprite, makes it invisible since we're always copying overlays anyway)
- makes it transfer the transform matrix and the pixel_x, pixel_y, layer, dir so there's a seamless transition between person and statue
- adds sanity checking so runtimes don't happen
- adds how much time there is left on the cooldown to transform again to the transform message
- also makes it so instead of the message for being turned to stone being in big massive red letters, since we're going to be clicking the button on purpose to do it anyway, just makes it a normal warning message
- fixes gargoyles losing health by changing their size while transformed, or just generally losing/gaining max health while transformed in general
- _**makes it so gargoyles can't click anything (move items on their person, etc, while transformed).**_ This was because I can think of a few different ways this could have been abused, so. If I can think of them, other people can, so I fixed that. Plus, statues really shouldn't be able to move at all. While they're statues, at least.
- adds flavour texts to examine more on gargoyles (ie, examine closer, double examine, etc)
- gargoyles no longer dust their owners when destroyed
- slicing apart a gargoyle will give a little alert first confirming if you want to do it
- BIG CHANGE, dunno if maintainers wanna chime in and say no: being a statue now heals you for brute/burn first (same slow rate as before), then (slower) tox loss, then (slower, again) clone loss, and then it slowly heals the statue structure's integrity over time (about 12 integrity per minute (which is really fucking slow))

also makes the code just that little bit more efficient by moving stuff around a bit

## Why It's Good For The Game

sanity, fixes and qols are always good

## A Port?

no

## Pre-Merge Checklist

- [x] This code did not runtime during testing.
- [x] You tested this on a local server.
- [x] You documented all of your changes.

## Changelog

:cl:
fix: fixed gargoyles having sprites that are odd, along with sanity checks and making the transition between living and statue seamless
/:cl:

### CODE REVIEW

**Comment by Cleric2145** [comment]
While you're fixing the gargoyle stuff, do you think you could take a look at a weird bug(?) I've noticed?

Steps:

1. Set sprite to 200% size (Experienced as a Xenobiological Slime Entity, unknown if it affects other races)
2. Add Normalizer wristband to loadout
3. Enter game, transform into statue
4. Remove Wristband
5. Unstatue

Result:
Player take full body/limb brute damage and some toxin damage. Unsure what could be causing or how difficult it could be to track down the cause. Don't stress too much about delaying this PR for it, just hoping it's a quick fix.

Expected:
No dmg ploz :(

**Comment by LunarFleet** [comment]
I truly appreciate the fixes being made, as I am not much of a coder myself, I just know enough code diving to cobble some basic stuff together. Though, y'could have likely brought this all up on the original PR and had it fixed then, but regardless, I thank you for your assistance~!

**Comment by Seris02** [comment]
> I truly appreciate the fixes being made, as I am not much of a coder myself, I just know enough code diving to cobble some basic stuff together. Though, y'could have likely brought this all up on the original PR and had it fixed then, but regardless, I thank you for your assistance~!

only started playing on splurt and saw the quirk yesterday or I would have XD

**Comment by Seris02** [comment]
> While you're fixing the gargoyle stuff, do you think you could take a look at a weird bug(?) I've noticed?
> 
> Steps:
> 
> 1. Set sprite to 200% size (Experienced as a Xenobiological Slime Entity, unknown if it affects other races)
> 2. Add Normalizer wristband to loadout
> 3. Enter game, transform into statue
> 4. Remove Wristband
> 5. Unstatue
> 
> Result: Player take full body/limb brute damage and some toxin damage. Unsure what could be causing or how difficult it could be to track down the cause. Don't stress too much about delaying this PR for it, just hoping it's a quick fix.
> 
> Expected: No dmg ploz :(

looked into it, fixed

**Comment by Cleric2145** [comment]
Ah...I can't help but feel like the addition of making the statue vulnerable will hurt the gargoyle trait. Actively slicing the statue makes sense, but accumulating damage dealt to it and dealing it afterward feels...kinda lame. If the gargoyle is statuing to heal a DoT (like toxin, fire, etc.) then the'll be damaged immediately afterward and need to restatue. One thing I find very useful about the perk is being able to statue while afk or during scrubber surges to avoid taking damage from clouds of chemicals nearby which might not affect a statue. Additionally, gargoyles can use statuing as emergency protection against depressurizations. A statue would not be expected to take damage from pressure until it was too extreme.

Given the amount of exceptions to damage a statue might have, I feel like this is more of a hindrance than not, as random damage can easily happen when you're locked into being stationary for an amount of time, whether random flying objects, careless people, or whatever else. The boon of being invincible is balanced by having to manage energy constantly by shifting, which locks them out for a minute or so. With the addition of being able to slice them, I feel like that will prevent gaming against sec/antags. Combined with being able to unscrew and drag them, this would already make statues fairly vulnerable if they try to abuse it.

Suggestion:
Perhaps damage incurred as a statue is converted into extra time before returning to normal and/or draining energy instead. This way, it's an active penalty rather than a delayed, frustrating one. I would think if a statue took damage and then remained a statue longer, that damage would heal and they could shift without taking damage?

**Comment by Seris02** [comment]
> I can't help but feel like the addition of making the statue vulnerable will hurt the gargoyle trait. Actively slicing the statue makes sense, but accumulating damage dealt to it and dealing it afterward feels...kinda lame. If the gargoyle is statuing to heal a DoT (like toxin, fire, etc.) then the'll be damaged immediately afterward and need to restatue. One thing I find very useful about the perk is being able to statue while afk or during scrubber surges to avoid taking damage from clouds of chemicals nearby which might not affect a statue. Additionally, gargoyles can use statuing as emergency protection against depressurizations. A statue would not be expected to take damage from pressure until it was too extreme.
> 
> Given the amount of exceptions to damage a statue might have, I feel like this is more of a hindrance than not, as random damage can easily happen when you're locked into being stationary for an amount of time, whether random flying objects, careless people, or whatever else. The boon of being invincible is balanced by having to manage energy constantly by shifting, which locks them out for a minute or so. With the addition of being able to slice them, I feel like that will prevent gaming against sec/antags. Combined with being able to unscrew and drag them, this would already make statues fairly vulnerable if they try to abuse it.
> 
> Suggestion: Perhaps damage incurred as a statue is converted into extra time before returning to normal and/or draining energy instead. This way, it's an active penalty rather than a delayed, frustrating one. I would think if a statue took damage and then remained a statue longer, that damage would heal and they could shift without taking damage?

wasn't an addition from this PR!
was already a thing from the first one
haven't made the gargoyles any more vulnerable than they already were, and I'm not really inclined to do anything about it tbh. It's a good way to stop abuse from it, since you already passively heal as a statue.
You are literally put into godmode when you're statued, it's impossible to do damage, so you'll only be healing, the only thing that will do damage is literally just breaking the statue, which would obviously do damage anyway.

Although, I will add it so that the object integrity slowly heals over time if the gargoyle inside is fully healed, which will make the damage when you come out get less and less. And I do mean **slowly**, so that you can't use it to just ignore security or something.

**Comment by LunarFleet** [comment]
> Thank you so much for your incredibly speedy work! You should check out some of the coding bounties! There's a solid $200 or so on offer right now.
> 
> I can't help but feel like the addition of making the statue vulnerable will hurt the gargoyle trait. Actively slicing the statue makes sense, but accumulating damage dealt to it and dealing it afterward feels...kinda lame. If the gargoyle is statuing to heal a DoT (like toxin, fire, etc.) then the'll be damaged immediately afterward and need to restatue. One thing I find very useful about the perk is being able to statue while afk or during scrubber surges to avoid taking damage from clouds of chemicals nearby which might not affect a statue. Additionally, gargoyles can use statuing as emergency protection against depressurizations. A statue would not be expected to take damage from pressure until it was too extreme.
> 
> Given the amount of exceptions to damage a statue might have, I feel like this is more of a hindrance than not, as random damage can easily happen when you're locked into being stationary for an amount of time, whether random flying objects, careless people, or whatever else. The boon of being invincible is balanced by having to manage energy constantly by shifting, which locks them out for a minute or so. With the addition of being able to slice them, I feel like that will prevent gaming against sec/antags. Combined with being able to unscrew and drag them, this would already make statues fairly vulnerable if they try to abuse it.
> 
> Suggestion: Perhaps damage incurred as a statue is converted into extra time before returning to normal and/or draining energy instead. This way, it's an active penalty rather than a delayed, frustrating one. I would think if a statue took damage and then remained a statue longer, that damage would heal and they could shift without taking damage?

Truthfully, the quirk was meant for roleplay; it was never meant to be used for mechanical gain, especially considering the fact that it is a 0 point cost quirk. That said, I was originally considering making it so the integrity of the statue slowly healed over time, similar to the petrified mob's HP; that may be worth doing, however... the cooldown to un-transform is only thirty seconds; if you start taking a bunch of damage, you can easily un-petrify and flee/resolve the situation.

**Comment by Cleric2145** [comment]
> wasn't an addition from this PR! was already a thing from the first one haven't made the gargoyles any more vulnerable than they already were, and I'm not really inclined to do anything about it tbh. It's a good way to stop abuse from it, since you already passively heal as a statue. You are literally put into godmode when you're statued, it's impossible to do damage, so you'll only be healing, the only thing that will do damage is literally just breaking the statue, which would obviously do damage anyway.
> 
> Although, I will add it so that the object integrity slowly heals over time if the gargoyle inside is fully healed, which will make the damage when you come out get less and less. And I do mean **slowly**, so that you can't use it to just ignore security or something.

Ah! Thank you! Sorry, I don't use github much so I'm not super familiar with reading the revisions.


> Truthfully, the quirk was meant for roleplay; it was never meant to be used for mechanical gain, especially considering the fact that it is a 0 point cost quirk. That said, I was originally considering making it so the integrity of the statue slowly healed over time, similar to the petrified mob's HP; that may be worth doing, however... the cooldown to un-transform is only thirty seconds; if you start taking a bunch of damage, you can easily un-petrify and flee/resolve the situation. Regardless, adding the integrity heal would likely be fine, however, otherwise 'buffing' the quirk's mechanical benefits likely will not fly~!

I didn't actually suggest any additional benefits past the object heal. Everything else was either speaking against the code I misread as doing something else, or suggesting alternative nerfs.

Honestly I think if the perk ever needed a nerf, then damage to the statue draining the energy would be the most logical IMO. If you get attacked and you statue, they will eventually perma-statue in essence and allow them to space you/chop you up. I think the perk is actually quite nicely balanced atm, as long as nobody tries to ultra-game it.

**Comment by BongaTheProto** [comment]
Might wanna put the new gargoyle icon in a modular file

**Comment by Cleric2145** [comment]
Could I request that the "Examine" verb be allowed while statued? Disabling it implies the gargoyle is completely blind and unaware of its surroundings, which doesn't seem to be the intent. While statued, there's already very little to do other than alt-tab.  Removing Examine just eliminates one of the last in-game things to do, and I don't really see a reason to disable it.

**Comment by Seris02** [comment]
> Could I request that the "Examine" verb be allowed while statued? Disabling it implies the gargoyle is completely blind and unaware of its surroundings, which doesn't seem to be the intent. While statued, there's already very little to do other than alt-tab. Removing Examine just eliminates one of the last in-game things to do, and I don't really see a reason to disable it.
> 
> Also, it feels like the energy regen rate dropped. I've been idling as a statue with small 5-10 minute bursts of activity and I dropped down to 18 energy. After quite a long time as a statue, I seem to be gaining energy at about 2/min. Is this just my perception, or did something break? I feels WAY slower than before. Idling for just shy of an hour should someone run out of energy isn't a very fun way to play, especially not even being able to read everybody's flavortext to help pass the time.

Ye, accidentally broke the regen, should not be merged yet
I'll push a quick fix for that while I try to fix another issue

**Comment by LunarFleet** [comment]
Something seems to be breaking the energy regeneration rate; three minutes of being a statue only restores about six points of energy. I have tested this on two different shifts, so it was not just the server lagging I am afraid. Obviously, that needs to be fixed, otherwise you'll end up spending most of your time as a statue~!

File: modular_splurt/code/datums/traits/neutral.dm [hunk 1]
```diff
@@ -567,29 +572,40 @@
...
 	if(transformed)
-		if(energy < 99.7)
-			energy += 0.3
-		H.heal_overall_damage(0.5,0.5)
-		H.adjustCloneLoss(-0.5)
-		H.adjustBruteLoss(-0.5)
-		H.adjustFireLoss(-0.5)
-
-	if(energy <= 0 && !transformed)
+		energy = min(energy + 0.03, 100)
```
Comment (BongaTheProto) [inline modular_splurt/code/datums/traits/neutral.dm:579]:
```suggestion
		energy = min(energy + 0.3, 100)
```

**Review by BongaTheProto — CHANGES_REQUESTED** [review]

**Comment by Seris02** [comment]
should be fine for merging now, still got one last thing, making it so you either can't rest or get up while as gargoyle, or making it so it puts you back to however you were (which doesn't seem to work for whatever reason, so probably gonna go with can't rest/stand)
because resting right now while being a statue breaks the rotation

**Comment by BongaTheProto** [comment]
> should be fine for merging now, still got one last thing, making it so you either can't rest or get up while as gargoyle, or making it so it puts you back to however you were (which doesn't seem to work for whatever reason, so probably gonna go with can't rest/stand) because resting right now while being a statue breaks the rotation

You could perhaps modularly edit `/mob/living/set_resting()` so that if you're inside a gargoyle statue with the quirk it just returns or else calls `. = ..()`

**Comment by Seris02** [comment]
> > should be fine for merging now, still got one last thing, making it so you either can't rest or get up while as gargoyle, or making it so it puts you back to however you were (which doesn't seem to work for whatever reason, so probably gonna go with can't rest/stand) because resting right now while being a statue breaks the rotation
> 
> You could perhaps modularly edit `/mob/living/set_resting()` so that if you're inside a gargoyle statue with the quirk it just returns, or else calls `. = ..()`

better
I just added a lying/lying_prev variable and set those to what it was, then update mobility

**Comment by Seris02** [comment]
ready for merging now

**Pull request merged**